### PR TITLE
Move to use Toasts and refactor CutMessage with CommonDelayableHook

### PR DESCRIPTION
**Pull request by ketal178**
# Move to use Toasts

## 更改内容 / Types of Changes
- 增强优化 Enhancement/optimization

## 由这个PR解决或关闭的Issues / Issues Fixed or Closed by This PR

-  

## 检查列表 / Checklist

- [x] 我的代码符合本仓库的代码规范 My code follows the code style of this project
- [x] 我已经测试了这些更改，它们可以正常工作，既不会破坏原有任何功能(或我可以解决)，也不会影响对旧版本的支持 I have tested the changes and verified that they work and don't break anything(as well as I can manage) or drop the support for previous versions.
- [x] 我已合并了对后续工作无意义的commit并确认不会对后续维护造成困扰 I have merged commits that are meaningless for follow-up work and confirmed that they will not cause damage to follow-up maintenance

### CODE REVIEW

**Review by qwq233 — CHANGES_REQUESTED** [review]
请合并几个commit 把delete啊那些本质上都是干一件事情(迁移的 合并一下

File: app/src/main/java/nil/nadph/qnotified/startup/StartupHook.java [hunk 1]
```diff
@@ -74,7 +75,7 @@ protected void afterHookedMethod(MethodHookParam param) throws Throwable {
                         app = (Context) fsApp.get(null);
                         execStartupInit(app, param.thisObject, null, false);
                     } catch (Throwable e) {
-                        log_e(e);
+                        Utils.log(e);
```
Comment (cinit) [inline app/src/main/java/nil/nadph/qnotified/startup/StartupHook.java:78]: Potential may cause crash

File: app/src/main/java/nil/nadph/qnotified/startup/StartupHook.java [hunk 2]
```diff
@@ -93,7 +94,7 @@ protected void afterHookedMethod(MethodHookParam param) throws Throwable {
         } catch (Throwable e) {
             if ((e + "").contains("com.bug.zqq")) return;
             if ((e + "").contains("com.google.android.webview")) return;
-            log_e(e);
+            Utils.log(e);
```
Comment (cinit) [inline app/src/main/java/nil/nadph/qnotified/startup/StartupHook.java:97]: The same above

**Review by cinit — CHANGES_REQUESTED** [review]

File: app/src/main/java/cc/ioctl/activity/ManageScriptsActivity.java [hunk 3]
```diff
@@ -89,10 +89,10 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
                     String path = uri.getPath();
                     try {
                         QNScriptManager.addScript(path);
-                        Utils.showToastShort(this, "添加完毕");
+                        Toasts.info(this, "添加完毕");
                     } catch (Exception e) {
                         log(e);
-                        Utils.showToastShort(this, "未知错误: " + e.getMessage());
+                        Toasts.info(this, "未知错误: " + e.getMessage());
```
Comment (NextAlone) [inline app/src/main/java/cc/ioctl/activity/ManageScriptsActivity.java:95]: error

File: app/src/main/java/cc/ioctl/activity/ManageScriptsActivity.java [hunk 4]
```diff
@@ -109,23 +109,23 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
                                 FileDescriptor fileDescriptor = parcelFileDescriptor.getFileDescriptor();
                                 String err = QNScriptManager.addScriptFD(fileDescriptor, scriptName);
                                 if (err.isEmpty()) {
-                                    Utils.showToastShort(this, "添加完毕");
+                                    Toasts.info(this, "添加完毕");
                                 } else {
-                                    Utils.showToastShort(this, err);
+                                    Toasts.info(this, err);
```
Comment (NextAlone) [inline app/src/main/java/cc/ioctl/activity/ManageScriptsActivity.java:114]: error

File: app/src/main/java/cc/ioctl/activity/ManageScriptsActivity.java [hunk 5]
```diff
@@ -109,23 +109,23 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
...
                                 }
                             }
                         } catch (Throwable e) {
                             log(e);
-                            Utils.showToastShort(this, "未知错误：" + e.getMessage());
+                            Toasts.info(this, "未知错误：" + e.getMessage());
```
Comment (NextAlone) [inline app/src/main/java/cc/ioctl/activity/ManageScriptsActivity.java:119]: error

File: app/src/main/java/nil/nadph/qnotified/startup/StartupHook.java (hunk 1 above)
Comment (ketal178) [reply app/src/main/java/nil/nadph/qnotified/startup/StartupHook.java:78]: ?离谱？？？

File: app/src/main/java/cc/ioctl/activity/ManageScriptsActivity.java [hunk 6]
```diff
@@ -109,23 +109,23 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
...
                         }
                     }
                     c.close();
                 }
             } else {
-                Utils.showToastShort(this, "内部错误");
+                Toasts.info(this, "内部错误");
```
Comment (NextAlone) [inline app/src/main/java/cc/ioctl/activity/ManageScriptsActivity.java:125]: error

File: app/src/main/java/cc/ioctl/activity/ManageScriptsActivity.java [hunk 7]
```diff
@@ -109,23 +109,23 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
...
             }
         } else {
-            Utils.showToastShort(this, "未知错误");
+            Toasts.info(this, "未知错误");
```
Comment (NextAlone) [inline app/src/main/java/cc/ioctl/activity/ManageScriptsActivity.java:128]: error

File: app/src/main/java/nil/nadph/qnotified/startup/StartupHook.java (hunk 1 above)
Comment (cinit) [reply app/src/main/java/nil/nadph/qnotified/startup/StartupHook.java:78]: 这个 `log_e()` 不要动，startup包内的类不应该访问startup包以外的类

**Review by NextAlone — CHANGES_REQUESTED** [review]

**Review by cinit — APPROVED** [review]

**Review by NextAlone — APPROVED** [review]

**Review by qwq233 — APPROVED** [review]

**Pull request merged**